### PR TITLE
Added article for Depth First Search

### PR DESCRIPTION
**Pull request by madihamallick**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes: #223

### CODE REVIEW

File: docs/graph-algorithms/depth-first-search.mdx [hunk 1]
```diff
@@ -0,0 +1,346 @@
+---
+title: Depth First Search
+slug: /graph-algorithms/depth-first-search
+---
+
+## Introduction
+
+Depth First Search or DFS uses stack data structure. The algorithm starts with an initial node of the Graph and goes deeper and deeper until it finds a node which has no children called the **dead node**. Then it starts backtracking from the dead node towards the most recent
```
Comment (vybhav72954) [inline docs/graph-algorithms/depth-first-search.mdx:8]: node **that** has.

that **lead** to an already

File: docs/graph-algorithms/depth-first-search.mdx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+## How does is work?
+
+Consider a six node graph where the number of edges are four:
+- Edge from `vertex A to vertex B`
+- Edge from `vertex B to vertex C`
+- Edge from `vertex A to vertex C`
+- Edge from `vertex D to vertex E`
+
+### The depth first traversal of the above will be as follows:
```
Comment (vybhav72954) [inline docs/graph-algorithms/depth-first-search.mdx:19]: These screenshots can be put up in a `1*5` table for better read

File: docs/graph-algorithms/depth-first-search.mdx [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+## Algorithm
+
+- Set status as ready state for each node in the Graph. 
+- Push the first node to the stack and set its status as waiting state.
+- Pop the top node and after processing set its status as processed state.
+- push on the stack all the neighbours of top node that are in the ready state.
+- change the status of the top node and set its status to waiting state.
+- Repeat the steps until the stack is empty.
+
+
```
Comment (vybhav72954) [inline docs/graph-algorithms/depth-first-search.mdx:36]: Extra newline, not required

File: docs/graph-algorithms/depth-first-search.mdx [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+</TabItem>
+
+<TabItem value="cpp">
+
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+void dfs(int source, vector <int> adj[],bool visited[]){
+    //visit the source
```
Comment (vybhav72954) [inline docs/graph-algorithms/depth-first-search.mdx:137]: // visit the source

File: docs/graph-algorithms/depth-first-search.mdx [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+    cout<<source<<" ";
+    //mark visited as true
```
Comment (vybhav72954) [inline docs/graph-algorithms/depth-first-search.mdx:139]: Missing whitespace

File: docs/graph-algorithms/depth-first-search.mdx [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+    visited[source] = true;
+    //Traverse all its adjacents
```
Comment (vybhav72954) [inline docs/graph-algorithms/depth-first-search.mdx:141]: Missing whitespace and please refactor other ones as well.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: docs/graph-algorithms/depth-first-search.mdx (hunk 6 above)
Comment (madihamallick) [reply docs/graph-algorithms/depth-first-search.mdx:141]: > Missing whitespace and please refactor other ones as well.

I'm sorry I didn't get it, where the whitespaces are missing ?

**Comment by madihamallick** [comment]
@vybhav72954 I have made the changes. Please review.

**Comment by madihamallick** [comment]
@HarshCasper Please review.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**